File: GeneratedEntityReaderClasses/Components/ItemPickUpperComponent.cs
namespace GeneratedNoitaClasses;
public class ItemPickUpperComponent : NoitaComponentBase,  iNoitaType<ItemPickUpperComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool is_in_npc { get; set; } = new();
	public NoitaBool pick_up_any_item_buggy { get; set; } = new();
	public NoitaBool is_immune_to_kicks { get; set; } = new();
	public NoitaUInt only_pick_this_entity { get; set; } = new();
	public NoitaBool drop_items_on_death { get; set; } = new();
	public NoitaCVector2<NoitaFloat> mLatestItemOverlapInfoBoxPosition { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		is_in_npc.Read(s);
		pick_up_any_item_buggy.Read(s);
		is_immune_to_kicks.Read(s);
		only_pick_this_entity.Read(s);
		drop_items_on_death.Read(s);
		mLatestItemOverlapInfoBoxPosition.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		is_in_npc.Write(s);
		pick_up_any_item_buggy.Write(s);
		is_immune_to_kicks.Write(s);
		only_pick_this_entity.Write(s);
		drop_items_on_death.Write(s);
		mLatestItemOverlapInfoBoxPosition.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ItemPickUpperComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class ItemPickUpperComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    is_in_npc:NoitaBool = field(default_factory=NoitaBool)
    pick_up_any_item_buggy:NoitaBool = field(default_factory=NoitaBool)
    is_immune_to_kicks:NoitaBool = field(default_factory=NoitaBool)
    only_pick_this_entity:NoitaUInt = field(default_factory=NoitaUInt)
    drop_items_on_death:NoitaBool = field(default_factory=NoitaBool)
    mLatestItemOverlapInfoBoxPosition:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.is_in_npc.Read(s)
        self.pick_up_any_item_buggy.Read(s)
        self.is_immune_to_kicks.Read(s)
        self.only_pick_this_entity.Read(s)
        self.drop_items_on_death.Read(s)
        self.mLatestItemOverlapInfoBoxPosition.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.is_in_npc.Write(s)
        self.pick_up_any_item_buggy.Write(s)
        self.is_immune_to_kicks.Write(s)
        self.only_pick_this_entity.Write(s)
        self.drop_items_on_death.Write(s)
        self.mLatestItemOverlapInfoBoxPosition.Write(s)

